### PR TITLE
Vb 1486 update entities to ensure they are not data classess

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
## What does this pull request do?

Make entity classes non data classes and add specific equals, toString and hashCode methods for each entity. This is to get around the Intellij warning of "The data class implementations of equals(), hashCode() and toString() are not recommended for JPA entities. They can cause severe performance and memory consumption issues."

## What is the intent behind these changes?
As recommended JPA entities should not use data class implementations of equals(), hashCode() and toString(). Also converted the entities from data classes to non data classes.
_Required._

### CODE REVIEW

**Review by AledWEvans — CHANGES_REQUESTED** [review]
please read comments

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/Visit.kt [hunk 1]
```diff
@@ -104,7 +104,6 @@ data class Visit(
   @Column
   @NaturalId(mutable = true)
   var applicationReference: String = ""
-    private set
```
[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/Visit.kt:107]: sorry this was not intended.. seems to pass without but I will revert this back

[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/Visit.kt:107]: no worries

**Comment by dhirajshettymoj** [comment]
Closing for now and instead raising a new merge request to main branch

**Comment by dhirajshettymoj** [comment]
Closing for now and instead raising a new merge request to main branch

**Pull request closed without merging**